services/src/datasets/external/gbif.rs

```diff
@@ -384,6 +587,106 @@ impl LayerCollectionProvider for GbifDataProvider {
             metadata: Default::default(),
         })
     }
+
+    async fn get_search_capabilities(&self) -> Result<SearchCapabilities> {
+        Ok(SearchCapabilities {
+            caps: vec![
+                SearchCapability {
+                    search_type: SearchType::FULLTEXT,
+                    autocomplete: true, // TODO have to see whether its too slow
+                    filters: None,
+                },
+                SearchCapability {
+                    search_type: SearchType::PREFIX,
+                    autocomplete: true, // TODO this should work fine with index
+                    filters: None, // TODO this is not needed in the current design since any filters applied are already part of the GBIF provider collection ID, so should we implement it now at all?
+                },
+            ],
+        })
+    }
+
+    async fn search(&self, search: SearchParameters) -> Result<LayerCollection> {
+        let collection = search.collection_id;
+        let selector = collection.0.split_once('/').map_or_else(
+            || collection.clone().0,
+            |(selector, _)| selector.to_string(),
+        );
+        let path = collection
+            .0
+            .split_once('/')
+            .map_or_else(|| "", |(_, path)| path);
+
+        let search_string = match search.search_type {
+            SearchType::FULLTEXT => format!("%{}%", search.search_string),
+            SearchType::PREFIX => format!("{}%", search.search_string),
+        };
+
+        let items = match selector.as_str() {
+            "datasets" => {
+                self.get_datasets_search_items(&search.limit, &search.offset, path, &search_string)
+                    .await?
+            }
+            "filter" => {
+                self.get_filter_search_items(&search.limit, &search.offset, path, &search_string)
+                    .await?
+            }
+            "select" => Self::get_select_items(path),
+            _ => vec![],
+        };
+
+        Ok(LayerCollection {
+            id: ProviderLayerCollectionId {
+                provider_id: GBIF_PROVIDER_ID,
+                collection_id: collection.clone(),
+            },
+            name: "GBIF".to_owned(),
+            description: "GBIF occurrence datasets".to_owned(),
+            items,
+            entry_label: None,
+            properties: vec![],
+        })
+    }
+
+    async fn autocomplete_search(&self, search: SearchParameters) -> Result<Vec<String>> {
```

Comment:
I would suggest to output an empty vec if the string is less then, e.g., 3 chars

Reply:
My plan is to try this out, how well it works for less than 3 chars. Probably it's not even consistent across queries how long this takes, so a timeout cutoff would probably make more sense

Reply:
I added a short (3s) timeout for the autocomplete queries in the GBIF provider. https://github.com/geo-engine/geoengine/pull/900/commits/c048c4924fe20323e57b6d6ae67f718a577b1c16

Reply:
Maybe add this to the config instead of hardcoding 3 seconds?
For slow machines, admins may decide that, e.g., 5 is ok.

Reply:
https://github.com/geo-engine/geoengine/pull/900/commits/a33cf37099682413e02e17e750501ef0d89231bc